modules/systemtests/src-test/org/ejbca/core/ejb/ca/caadmin/RenewCATest.java

```diff
@@ -97,6 +99,84 @@ public void test01renewCA() throws Exception {
+        cryptoTokenManagementSession.createKeyPair(internalAdmin, caToken.getCryptoTokenId(), sNextKeyAlias, com.keyfactor.util.keys.token.KeyGenParams.builder("prime256v1").build());
+
+        // To get EJBCA to renew a CA with a different key algorithm, we need to:
+        //   1. set up the signing algorithm in the CA's token to support EC, 
+        //   2. ensure the certificate profile has an appropriate signature algorithm to support EC.
+        
+        // Set the signature algorithm of the token
+        caToken.setSignatureAlgorithm( AlgorithmConstants.SIGALG_SHA256_WITH_ECDSA);
+        
+        // Need to set the Certificate Profile to use ECDSA based signature algorithm
+        // Lets copy the current profile, change it, and save a new profile (as we can't edit the current profile)
+        int iCP = info.getCertificateProfileId();
+        org.cesecore.certificates.certificateprofile.CertificateProfile cpCA = certificateProfileSession.getCertificateProfile(iCP);
+        cpCA.setSignatureAlgorithm( AlgorithmConstants.SIGALG_SHA256_WITH_ECDSA);
+        iCP = certificateProfileSession.addCertificateProfile(internalAdmin, "TESTRENEWALWITHEC", cpCA);
+        // Update the CA
+        info.setCertificateProfileId(iCP);
+        caSession.editCA(internalAdmin, info);
+
+        // We are all set and now ready to renew the CA
+        caAdminSession.renewCA(internalAdmin, info.getCAId(), sNextKeyAlias, null, /*CreateLinkCert*/true);
+
+        // Remove the certificate profile we just created.
+        certificateProfileSession.removeCertificateProfile(internalAdmin, "TESTRENEWALWITHEC");
+        
+        // Let check the CA's new certificate has the ECDSA based signing algorithm
+        X509CAInfo newinfo = (X509CAInfo) caSession.getCAInfo(internalAdmin, "TEST");
+        X509Certificate newcert = (X509Certificate) newinfo.getCertificateChain().iterator().next();
+        String sNewSigAlg = ((X509Certificate)newcert).getSigAlgName();
+        assertTrue( "Previous Signing Algorith was "+sPreviousSigAlg+" and new Signing Algorithm was "+sNewSigAlg+". Was expecting it to be ECDSA based.",
+                sNewSigAlg.contains("ECDSA"));
+       
+        // Check the Link certificate was signed using the previous Signing Algorithm
+        byte[] linkCertificateAfterRenewal1Bytes = caAdminSession.getLatestLinkCertificate(newinfo.getCAId());
+        assertTrue("There is no available link certificate after CA renewal with EC key", linkCertificateAfterRenewal1Bytes != null);
+        @SuppressWarnings("deprecation")
+        X509Certificate linkCertificateAfterRenewal1 = (X509Certificate) com.keyfactor.util.CertTools.getCertfromByteArray(linkCertificateAfterRenewal1Bytes);
+        assertTrue("The Link certificate should be signed by the CA's previous signing algorithm, not "+linkCertificateAfterRenewal1.getSigAlgName(),
+                linkCertificateAfterRenewal1.getSigAlgName().equalsIgnoreCase(sPreviousSigAlg) );
+
```

Comment:
Add some more tests on the link certificate. subject/issuer, keyIDs, other extensions you expect?

Reply:
Additional assertions added to check Subject and Issuer DNs written intp the Link certificate, as well as checking the changes to the CA Token and certificate profile are reverted. I don't think it necessary to check KeyIDs and other things that should otherwise be tested by existing CA Renewal test cases. 

I've also added another testcase for checking a CA key algorithm change along with a SubjectDN change. This is important as the main code change did surround an "if (subjectDNWillBeChanged)" statement.  The new testcase shows no issues.